src/AsyncUtils.js

```diff
@@ -63,7 +69,7 @@ export class AsyncQueue {
         this.createPromise();
       }
     } finally {
```

Comment:
do we still need this finally block? it seems like calling `.return()` is never the right thing to do, so we may as well not support it? or is this for error handling?